src/Knp/Snappy/Pdf.php

```diff
@@ -29,19 +32,15 @@ public function __construct($binary = null, array $options = array(), array $env
      */
     protected function handleOptions(array $options = array())
     {
-        $this->hasHtmlHeader = false;
-        $this->hasHtmlFooter = false;
-        
-        if ($this->isFileHeader($options) && !$this->isFile($options['header-html'])) {
-            $options['header-html'] = $this->createTemporaryFile($options['header-html'], 'html');
-        }
-
-        if ($this->isFileFooter($options) && !$this->isFile($options['footer-html'])) {
-            $options['footer-html'] = $this->createTemporaryFile($options['footer-html'], 'html');
-        }
-
-        if ($this->isFileCover($options) && !$this->isFile($options['cover'])) {
-            $options['cover'] = $this->createTemporaryFile($options['cover'], 'html');
+        foreach ($options as $option => $value) {
+            if(null === $value){
```

Comment:
Hm, I have rechecked the [build command method](https://github.com/alexisduque/snappy/blob/master/src/Knp/Snappy/AbstractGenerator.php#L372) and all the options with ```null``` are skipped there, so my answer would be - no, it would not break any options but maybe I am wrong.